### PR TITLE
[ENH] Update JSONLD examples with `@context` and outputs from `bagel derivatives`

### LINKED ISSUE #33: Update examples with pipeline metadata
- [x] Add two new JSONLDs for (1) pheno-and-derivatives-only, and (2) pheno-bids-derivatives CLI outputs
- [x] Add a reference imaging bagel from Nipoppy
- [ ] Update examples of API results

### PR DESCRIPTION
**Pull request by alyssadai**
- Closes #33 
- Related to neurobagel/bagel-cli#349

Changes proposed in this pull request:

- Existing JSONLDs regenerated with new `@context` incl. derivatives terms
- New examples added for:
  - Nipoppy-compliant processing status TSV 
  - Output of `bagel derivatives` command w/ TSV & pheno-only JSONLD (from this repo)
  - Output of `bagel derivatives` command w/ TSV & pheno- and bids-containing JSONLD (from this repo)

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: data-upload/example_synthetic.jsonld [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
   "@context": {
-    "nb": "http://neurobagel.org/vocab/",
-    "snomed": "http://purl.bioontology.org/ontology/SNOMEDCT/",
-    "nidm": "http://purl.org/nidash/nidm#",
     "cogatlas": "https://www.cognitiveatlas.org/task/id/",
+    "nb": "http://neurobagel.org/vocab/",
     "ncit": "http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#",
+    "nidm": "http://purl.org/nidash/nidm#",
+    "snomed": "http://purl.bioontology.org/ontology/SNOMEDCT/",
+    "np": "https://github.com/nipoppy/pipeline-catalog/tree/main/processing",
```
Comment (surchs) [inline data-upload/example_synthetic.jsonld:8]: why the change in order? just the output of generate_context()?

Comment (surchs) [reply data-upload/example_synthetic.jsonld:8]: Also: this jsonld doesn't actually have any pipeline instances, right? Why is `np` then in the context? Do we just slap the whole context on no matter if we need it?

**Review by surchs — APPROVED** [review]
Looks good, and thanks @alyssadai 

I have one question (but I think I know the answer), and :cook:

File: data-upload/example_synthetic.jsonld (hunk 1 above)
Comment (alyssadai) [reply data-upload/example_synthetic.jsonld:8]: > why the change in order? just the output of generate_context()?

Yes, it's just the (arbitrary) order in which the namespaces appear in the list `ALL_NAMESPACES`:
```bash
ALL_NAMESPACES = [COGATLAS, NB, NCIT, NIDM, SNOMED, NP]
```

> Also: this jsonld doesn't actually have any pipeline instances, right? Why is np then in the context? Do we just slap the whole context on no matter if we need it?

That's the idea, and that's what we've been doing so far. We only have one `generate_context()`, and it includes everything we model, grabbed from `models.py`. If we wanted the `@context` to be jsonld-specific (i.e., different for the output of every CLI command), the logic of `generate_context()` would need to be a lot more complex.

Anyways, since most subject attributes are optional, even if we include only e.g., phenotypic-related terms/namespaces in the `@context` for a pheno-only JSONLD, it does not guarantee those terms will appear in every JSONLD generated by `bagel pheno` - rather, it would depend on the dataset. To avoid these discrepencies, I think it still makes sense to have all namespaces/terms we support in the `@context`.

This also makes it much easier to test if the context we generate captures everything needed, since we expect it to be static.
